src/AppBundle/Controller/ActivityController.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+
+/**
+ * Class ActivityController
+ *
+ * @Route("/activity")
+ */
+class ActivityController extends Controller
+{
+    /**
+     * @Route("/", name="activity")
+     * @Template()
+     */
+    public function indexAction(Request $request)
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $filter = $this->get('app.activity_extractor_factory')->create();
+
+        if ($request->get('project_id')) {
```

Comment:
Now the `activity` route is closed by a `access_controll` rule. Also here should be a validation of the parameters too.